src/lp/empty_row.jl

```diff
@@ -1,21 +1,71 @@
-# TODO: this is redundant with forcing constraint checks
-#   => an empty row is automatically redundant or infeasible
+@doc raw"""
+    RemoveEmptyRow <: AbstractRule
 
+Eliminate a single row if it is empty, i.e. all variable coefficients are zero.
+In particular, consider row $i$ given as
+```math
+l^{r}_{i} ≤ \sum_{j} a_{i, j} x_{j} ≤ u^{r}_{i}.
+```
+We eliminate row $i$ if, for each variable index $j$,
+``math
+| a_{i,j} | ≤ ϵ
+```
+for prescribed tolerance $ϵ$.
+
+## Presolve
+
+If row $i$ is empty, the corresponding constraint is completely removed from
+the problem.
+
+## Infeasibility checks
+
+The problem is infeasible if row $i$ is empty and either 1) its
+lower bound is positive ($l^{r}_{i} > ϵ$) or 2) its upper bound is
+negative ($u^{r}_{i} < -ϵ$). In these two cases, a Farkas infeasibility
+certificate is given by setting $y$ to all zeros except for either
+$y\_lower_{i} = 1$ or $y\_upper_{i} = 1$, respectively.
+
+## Postsolve
+
+Sets the dual variable $y\_lower_{i} = \max\{0,y\}$ for the greater-than
+constraint and the dual variable $y\_upper_{i} = \max\{0,-y\}$ for the
+less-than constraint.
```

Comment:
Small clarification.
Unless row `i` is infeasible, both dual multipliers `y_l` and `y_u` are zero: this corresponds to the reduction `EmptyRow(i, zero(T))`, which is created in the non-infeasible case.

Thus, the `y_l = max(0, y)`, `y_u = max(0, -y)` in the post-solve only yield non-zero values if the row was infeasible.
My rationale for initially doing so was to have "status-agnostic" reductions and postsolve.

Reply:
Updated the doc, let me know what you think.